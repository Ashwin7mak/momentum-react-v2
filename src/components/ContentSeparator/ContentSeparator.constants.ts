const CLASS_PREFIX = 'md-content-separator';

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { CLASS_PREFIX, STYLE };
