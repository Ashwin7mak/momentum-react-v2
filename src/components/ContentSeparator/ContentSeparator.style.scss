.md-content-separator-wrapper {
  color: var(--mds-color-theme-text-secondary-normal);
  margin: 0.25rem 0;
  display: flex;
  flex-direction: row;
  &[data-gradient="true"] {
    &:after {
      border-bottom: none;
      height: 1px;
      background: linear-gradient(90deg, var(--mds-color-theme-gradientdivider-normal-0) -2.27%, var(--mds-color-theme-gradientdivider-normal-1) 21.16%, var(--mds-color-theme-gradientdivider-normal-1) 73.47%, var(--mds-color-theme-gradientdivider-normal-0) 102.35%);
    }
  }

  &:after {
    content: '';
    flex: 1 1;
    border-bottom: 1px solid var(--mds-color-theme-outline-primary-normal);
    margin: auto;
  }
}
