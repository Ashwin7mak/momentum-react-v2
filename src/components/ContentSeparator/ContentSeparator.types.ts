import { ReactNode } from 'react';

export interface Props {
  /**
   * className custom class to be able to override the content separators's CSS
   */
  className?: string;
  /**
   * The child components of this component.
   */
  children?: ReactNode;
}
