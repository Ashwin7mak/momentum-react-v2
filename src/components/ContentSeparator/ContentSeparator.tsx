import React from 'react';
import classnames from 'classnames';

import './ContentSeparator.style.scss';
import { Props } from './ContentSeparator.types';
import { STYLE } from './ContentSeparator.constants';

const ContentSeparator: React.FC<Props> = (props: Props) => {
  const { className, children } = props;

  return (
    <div className={STYLE.wrapper}>
      <div className={classnames(className)}>{children}</div>
    </div>
  );
};

/**
 * ContentSeparator with text/other component in the middle
 */

export default ContentSeparator;
