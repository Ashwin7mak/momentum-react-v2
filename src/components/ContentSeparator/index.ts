import { default as ContentSeparator } from './ContentSeparator';
import { Props } from './ContentSeparator.types';

export type ContentSeparatorProps = Props;

export default ContentSeparator;
