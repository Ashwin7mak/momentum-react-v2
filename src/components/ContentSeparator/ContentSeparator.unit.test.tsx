import ContentSeparator from '.';
import { mount } from 'enzyme';
import React from 'react';

describe('ContentSeparator', () => {
  describe('snapshot', () => {
    it('should match snapshot when text is used', () => {
      const container = mount(<ContentSeparator>hello</ContentSeparator>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot when another component is used', () => {
      const container = mount(
        <ContentSeparator>
          <div>
            <div>Some nested text</div>
          </div>
        </ContentSeparator>
      );

      expect(container).toMatchSnapshot();
    });
  });

  describe('props', () => {
    it('should render with default props', () => {
      const container = mount(<ContentSeparator>hello</ContentSeparator>);

      expect(container.find('li').prop('className')).toEqual('md-content-separator-wrapper');
      expect(container.find('li').prop('data-gradient')).toBeUndefined();
    });

    it('should render with correct props', () => {
      const container = mount(
        <ContentSeparator className="custom-class" gradient>
          <div>
            <div>Some nested text</div>
          </div>
        </ContentSeparator>
      );

      expect(container.find('li').prop('className')).toEqual(
        'custom-class md-content-separator-wrapper'
      );
      expect(container.find('li').prop('data-gradient')).toEqual(true);
    });
  });
});
