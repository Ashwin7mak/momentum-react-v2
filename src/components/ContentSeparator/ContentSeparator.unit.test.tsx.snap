// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentSeparator snapshot should match snapshot when another component is used 1`] = `
<ContentSeparator>
  <div
    className="md-content-separator-wrapper"
  >
    <div
      className=""
    >
      <div>
        <div>
          Some nested text
        </div>
      </div>
    </div>
  </div>
</ContentSeparator>
`;

exports[`ContentSeparator snapshot should match snapshot when text is used 1`] = `
<ContentSeparator>
  <div
    className="md-content-separator-wrapper"
  >
    <div
      className=""
    >
      hello
    </div>
  </div>
</ContentSeparator>
`;
